ui/test/services/bom_bar_client_test.dart

```diff
@@ -24,27 +24,45 @@ void main() {
     group('Projects', () {
       const projectId = 'projectId';
 
-      group('Get all projects', () {
-        test('queries projects', () async {
+      group('Get projects based on search query', () {
+        test('queries for all projects with no search fragment', () async {
           server.respondJson({
             'results': [
-              {'id': projectId}
+              {'id': projectId, 'title': 'ProjectA'},
+              {'id': projectId, 'title': 'ProjectB'}
             ]
           });
-          final projects = await client.getProjects();
+          final projects = await client.findProjectsBySearchFragment('');
+          final request = server.requests.first;
+          expect(request.method, 'GET');
+          expect(request.path,
+              BomBarClient.baseUrl.resolve('/projects/').toString());
+          expect(projects.length, 2);
+          expect(projects[0].id, projectId);
+        });
 
+        test('queries for projects matching search fragment', () async {
+          server.respondJson({
+            'results': [
+              {'id': projectId, 'title': 'ProjectA'}
+            ]
+          });
+          final projects =
+              await client.findProjectsBySearchFragment('ProjectA');
```

Comment:
Suggestion is to create a (local) `const` for the fragment, and use it in the Act and Assert statements of this test.

(It might also be clearer if you use a value like "fragment" for that constant.)